src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -235,6 +232,52 @@ private static bool StartsWithOrdinal(this StringBuilder builder, string value)
         }
 
         /// <summary>
+        /// Returns true if the given StringBuilder starts with the given value at the given index.
+        /// </summary>
+        /// <param name="value">The string to compare against the start of the StringBuilder.</param>
+        /// <param name="index">The starting index to look for a match.</param>
+        internal static bool StartsWithOrdinal(this string source, string value, int index = 0)
+        {
+            if (value == null || source.Length < value.Length || index < 0)
+                return false;
+
+            for (int i = 0; i < value.Length; i++)
+            {
```

Comment:
This seems to be doing full string ordinal comparison. Should the loop start with index?